### PR TITLE
disk_block_cache: Make disk cache errors nonfatal

### PR DESCRIPTION
**Pull request by jzila**
First, ensure that a failure to load a version file isn't fatal for the disk cache.

Then, even if the disk cache fails to load, start KBFS anyway.

### CODE REVIEW

File: libkbfs/disk_block_cache.go [hunk 1]
```diff
@@ -256,27 +260,38 @@ func getVersionedPathForDiskCache(dirPath string) (versionedDirPath string,
...
-		if version < currentDiskCacheVersion {
-			return "", errors.WithStack(
-				InvalidVersionError{fmt.Sprintf("New disk cache version."+
-					" Delete the existing disk cache at path: %s", dirPath)})
-		}
-		// Existing disk cache version is newer than we expect for this client.
-		// This is an error since our client won't understand its format.
-		if version > currentDiskCacheVersion {
-			return "", errors.WithStack(OutdatedVersionError{})
+		if err != nil {
```
Comment (strib) [inline libkbfs/disk_block_cache.go]: It would be good to log something in all three of these cases.

File: libkbfs/init.go [hunk 2]
```diff
@@ -616,11 +616,13 @@ func doInit(ctx Context, params InitParams, keybaseServiceCn KeybaseServiceCn,
 		log.Warning("Could not enable disk limiter: %+v", err)
 		return nil, err
 	}
+	goctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```
Comment (strib) [inline libkbfs/init.go]: Nit: why "goctx"?  It's not used in a goroutine.  Maybe `ctxWithTimeout` or `ctx10s`?

Comment (jzila) [inline libkbfs/init.go]: `ctx` was taken by something that isn't a `context.Context`, so I was just going for a name that differentiated it. I'll change it to `ctx10s`.

File: libkbfs/disk_block_cache.go (hunk 1 above)
Comment (jzila) [inline libkbfs/disk_block_cache.go]: K I'll plumb the logger through.

File: libkbfs/init.go (hunk 2 above)
Comment (strib) [inline libkbfs/init.go]: Ah I see, the "go" referred to "golang", not "goroutine".  Cool.  `ctx10s` still seems better to me, thanks.

**Pull request merged**